lib/matplotlib/tests/test_subplots.py

```diff
@@ -30,13 +30,13 @@ def check_visible(axs, x_visible, y_visible):
     def tostr(v):
         return "invisible" if v else "visible"
 
-    for ax, vx, vy in zip(axs, x_visible, y_visible):
+    for i, (ax, vx, vy) in enumerate(zip(axs, x_visible, y_visible)):
         for l in ax.get_xticklabels() + [ax.get_xaxis().offsetText]:
             assert l.get_visible() == vx, \
-                    "X axis was incorrectly %s" % (tostr(vx))
+                    f"X axis #{i} is incorrectly {tostr(vx)}"
```

Comment:
Would just go for
~~~
            visibility = "invisible" if vx else "visible"
            assert l.get_visible() == vx, \
                "x-axis of Axes {i} was incorrectly {visibility}"
~~~

Reply:
but then you need to do the same for y.  rephrased the thing.